internal/sessiontest/requestor_test.go

```diff
@@ -42,67 +68,161 @@ func processOptions(options ...sessionOption) sessionOption {
 	return opts
 }
 
-func requestorSessionHelper(t *testing.T, request interface{}, client *irmaclient.Client, options ...sessionOption) *requestorSessionResult {
-	if client == nil {
-		var handler *TestClientHandler
-		client, handler = parseStorage(t)
-		defer test.ClearTestStorage(t, handler.storage)
-	}
+func (o sessionOption) enabled(opt sessionOption) bool {
+	return o&opt > 0
+}
 
-	opts := processOptions(options...)
-	if opts&sessionOptionReuseServer == 0 {
-		StartIrmaServer(t, opts&sessionOptionUpdatedIrmaConfiguration > 0, "")
-		defer StopIrmaServer()
+// startServer ensures that an IRMA server or library is running, if and when required, as specified
+// by the parameters:
+// - If irmaServer is not nil or sessionOptionReuseServer is enabled, this function does nothing.
+// - Otherwise an IRMA server or library is started, depending on the type of conf.
+func startServer(t *testing.T, opts sessionOption, irmaServer *IrmaServer, conf interface{}) (stoppable, interface{}, bool) {
+	if irmaServer != nil {
+		if opts.enabled(sessionOptionReuseServer) {
+			require.FailNow(t, "either specify irmaServer or sessionOptionReuseServer")
+			return nil, nil, false
+		}
+		if conf != nil {
+			require.FailNow(t, "either specify irmaServer or conf")
+			return nil, nil, false
+		}
+		return irmaServer, nil, false
+	}
+	if opts.enabled(sessionOptionReuseServer) {
+		return nil, nil, false
```

Comment:
Currently a configuration can be  passed to the `doSession` function in combination with `sessionOptionReuseServer`. However, the code will only work with the JWT configuration as no configuration is returned here.

This is fine as we currently don't have any test where we need to use a different configuration. So I suggest we error out here if a configuration is passed and add a comment where `sessionOptionReuseServer` is defined that this option can only be used with JWT configurations.

Reply:
Thanks!

I've just noticed that here and also above yo return after the require.FailNow. If I get FailNow correctly, the return statements will never be triggered and can be removed.